study1/week4/MaxAreaOfIsland.java

```diff
@@ -0,0 +1,52 @@
+class Solution {
+    public int maxAreaOfIsland(int[][] grid) {
+        int maxArea = 0;
+        int m = grid.length;
+        int n = grid[0].length;
+
+        for (int i = 0; i < m; i++) {
+            for (int j = 0; j < n; j++) {
+                if (grid[i][j] == 1) {
+                    int area = 0;
+                    Queue<int[]> queue = new LinkedList<>();
+                    queue.offer(new int[]{i, j});
+                    grid[i][j] = 0; // mark the visited field as 0 to avoid dobble check
+
+                    while (!queue.isEmpty()) {
+                        int[] current = queue.poll();
+                        int x = current[0];
+                        int y = current[1];
+                        area++;
+
+                        // check 4 directions (up, down, left, right)
+                        int[][] directions = {{-1, 0}, {1, 0}, {0, -1}, {0, 1}};
```

Comment:
So this will simplify code a bit, but in line 27, you're potentially comparing where you don't need to. For `up`, you only need to compare if `row - 1 >=0`, but you are comparing unnecessarily four times with the values you don't need to compare.